scripts/microshift-cleanup-data.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+SCRIPT_NAME=$(basename $0)
+# The 'openshift-ovn-kubernetes' namespace must be in the end of the list
+# to allow for pod deletion when MicroShift is stopped
+PODS_NS_LIST=(openshift-service-ca openshift-ingress openshift-dns openshift-storage openshift-ovn-kubernetes)
+FULL_CLEAN=true
+
+function usage() {
+    echo "Stop all MicroShift services, also cleaning their data"
+    echo ""
+    echo "Usage: ${SCRIPT_NAME} <--all | --ovn>"
+    echo "   --all   Clean all MicroShift and OVN data"
+    echo "   --ovn   Clean OVN data only"
+    exit 1
+}
+
+# Parse command line
+case $1 in
+--all)
+    ;;
+--ovn)
+    FULL_CLEAN=false
+    ;;
+*)
+    usage
+    ;;
+esac
+
+# Exit if the current user is not 'root'
+if [ $(id -u) -ne 0 ] ; then
+    echo "The '${SCRIPT_NAME}' script must be run with the 'root' user privileges"
+    exit 0
+fi
+
+if ${FULL_CLEAN} ; then
+    echo "DATA LOSS WARNING: Do you wish to stop and clean ALL MicroShift data AND cri-o container workloads?"
+    select yn in "Yes" "No"; do
+        case "${yn}" in
+        Yes) 
+            break
+            ;;
+        *)
+            echo "Aborting cleanup" 
+            exit 0
+            ;;
+        esac
+    done
+fi
+
+echo Stopping MicroShift services
+for service in microshift microshift-etcd ; do
+    systemctl stop --now   ${service} 2>/dev/null
+    systemctl reset-failed ${service} 2>/dev/null
+done
+
+if ${FULL_CLEAN} ; then
+    echo Disabling MicroShift services
+    systemctl disable microshift 2>/dev/null
+fi
+pkill -9 --exact microshift
+pkill -9 --exact microshift-etcd
+
+if ${FULL_CLEAN} ; then
+    echo Removing crio container and image storage
+    crio wipe -f &>/dev/null
+    systemctl restart crio
+else
+    echo Removing MicroShift pods
+    for i in ${!PODS_NS_LIST[@]}; do
+        ns=${PODS_NS_LIST[$i]}    
+
+        retries=5
+        while [ ${retries} -gt 0 ] ; do
+            ocp_pods=$(crictl pods --namespace ${ns} -q)
+            if [ "$(echo ${ocp_pods} | wc -w)" -eq 0 ] ; then
+                break
+            fi
+            crictl rmp -f ${ocp_pods} &>/dev/null
+            retries=$(( retries - 1 ))
+        done
+    done
+fi
+
+echo Killing conmon, pause and OVN processes
+ovs-vsctl del-br br-int 2>/dev/null
+for pname in conmon pause ovn-controller ovn-northd ovsdb-server ; do
```

Comment:
`pkill conmon` would kill all conmon process for all microshift containers.

from crio doc:
```
Each container is monitored by a separate conmon process. The conmon process holds the pty of the PID1 of the container process. It handles logging for the container and records the exit code for the container process
```

Reply:
Are you saying we should not kill conmon processes? At this time of the script, either all the crio pods/images have been wiped off, or every pod has been stopped. Does it make a difference?

Reply:
I see, I thought the non-ovn pods are still running at this stage, but it is not.